networkx/generators/classic.py

```diff
@@ -186,9 +186,9 @@ def barbell_graph(m1, m2, create_using=None):
     G.name = "barbell_graph(%d,%d)" % (m1, m2)
 
     # connecting path
-    G.add_nodes_from([v for v in range(m1, m1 + m2 - 1)])
+    G.add_nodes_from(v for v in range(m1, m1 + m2 - 1))
     if m2 > 1:
-        G.add_edges_from([(v, v + 1) for v in range(m1, m1 + m2 - 1)])
+        G.add_edges_from((v, v + 1) for v in range(m1, m1 + m2 - 1))
```

Comment:
This could just be

    from networkx.utils import pairwise
    G.add_edges_from(pairwise(range(...)))